echopype/clean/api.py

```diff
@@ -74,3 +85,206 @@ def remove_noise(ds_Sv, ping_num, range_sample_num, noise_max=None, SNR_threshol
+
+    Returns
+    -------
+    xr.DataArray
+        A DataArray consisting of a mask for the Sv data, wherein True values signify
+        samples that are free of noise.
+    """
+
+    # Our goal is to have a mask where True represents samples that are NOT impulse noise.
+    # So, we negate the obtained mask.
+
+    if method == "ryan":
+        impulse_mask_ryan = _find_impulse_mask_ryan(source_Sv, desired_channel, m, n, thr)
+        noise_free_mask = ~impulse_mask_ryan
+    elif method == "ryan_iterable":
+        impulse_mask_ryan_iterable = _find_impulse_mask_ryan_iterable(
+            source_Sv, desired_channel, m, n, thr
+        )
+        noise_free_mask = ~impulse_mask_ryan_iterable
+    elif method == "wang":
+        impulse_mask_wang = _find_impulse_mask_wang(
+            source_Sv, desired_channel, thr, erode, dilate, median
+        )
+        noise_free_mask = ~impulse_mask_wang
+    else:
+        raise ValueError(f"Unsupported method: {method}")
+
+    return noise_free_mask
+
+
+def get_attenuation_mask(
+    source_Sv: Union[xr.Dataset, str, pathlib.Path],
+    desired_channel: str,
+    mask_type: str = "ryan",
+    **kwargs,
+) -> xr.DataArray:
+    """
+    Create a mask based on the identified signal attenuations of Sv values at 38KHz.
+    This method is based on:
+    Ryan et al. (2015) ‘Reducing bias due to noise and attenuation in
+        open-ocean echo integration data’, ICES Journal of Marine Science,
+        72: 2482–2493.
+    and,
+    Ariza et al. (2022) 'Acoustic seascape partitioning through functional data analysis',
+    Journal of Biogeography, 00, 1– 15. https://doi.org/10.1111/jbi.14534
+
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data to create a mask for,
+        else it specifies the path to a zarr or netcdf file containing
+        a Dataset. This input must correspond to a Dataset that has the
+        coordinate ``channel`` and variables ``frequency_nominal`` and ``Sv``.
+    desired_channel: the Dataset channel to be used for identifying the signal attenuation.
+    mask_type: str with either "ryan" or "ariza" based on the
+                preferred method for signal attenuation mask generation
+    Returns
+    -------
+    xr.DataArray
+        A DataArray containing the mask for the Sv data. Regions satisfying the thresholding
+        criteria are filled with ``True``, else the regions are filled with ``False``.
+
+    Raises
+    ------
+    ValueError
+        If neither ``ryan`` or ``azira`` are given
+
+    Notes
+    -----
+
+
+    Examples
+    --------
+
+    """
+    assert mask_type in ["ryan", "ariza"], "mask_type must be either 'ryan' or 'ariza'"
+    selected_channel_Sv = source_Sv.sel(channel=desired_channel)
+    Sv = selected_channel_Sv["Sv"].values
+    r = source_Sv["echo_range"].values[0, 0]
```

Comment:
Let's do label-aware processing in the functions and not extract just the data to be numpy array, to take advantage of distributed computing when possible.